src/components/FullPageOfflineView.js

```diff
@@ -0,0 +1,60 @@
+import React from 'react';
+import PropTypes from 'prop-types';
+import {View} from 'react-native';
+import {withOnyx} from 'react-native-onyx';
+import Icon from './Icon';
+import Text from './Text';
+import * as Expensicons from './Icon/Expensicons';
+import themeColors from '../styles/themes/default';
+import styles from '../styles/styles';
+import withLocalize, {withLocalizePropTypes} from './withLocalize';
+import ONYXKEYS from '../ONYXKEYS';
+import FullScreenLoadingIndicator from './FullscreenLoadingIndicator';
+import compose from '../libs/compose';
+
+const propTypes = {
+    /** Child elements */
+    children: PropTypes.node.isRequired,
+
+    isLoading: PropTypes.bool.isRequired,
+
+    error: PropTypes.string,
+
+    /** Props to fetch translation features */
+    ...withLocalizePropTypes,
+};
+
+const FullPageOfflineView = (props) => {
+    if (ONYXKEYS.NETWORK.isOffline) {
+        return (
+            <View
+                style={[styles.flex1, styles.alignItemsCenter, styles.justifyContentCenter]}
+            >
+                <Icon
+                    src={Expensicons.OfflineCloud}
+                    fill={themeColors.offline}
+                    width={50}
+                    height={50}
+                />
+                <Text style={[styles.h1]}>{props.translate('reportActionCompose.youAppearToBeOffline')}</Text>
+            </View>
+        );
+    }
+
+    if (props.isLoading) {
+        return <FullScreenLoadingIndicator />;
+    }
+
+    return props.children;
+};
+
+FullPageOfflineView.propTypes = propTypes;
+
+export default compose(
+    withLocalize,
+    withOnyx({
```

Comment:
Should use `withNetwork` instead:

```js
import {withNetwork} from './OnyxProvider';
```

That uses one shared Onyx subscription and the React context API under the hood, so it's much more performant.